x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/session/EsqlResolvedIndexExpressionIT.java

```diff
@@ -13,29 +13,28 @@
 import java.util.Map;
 import java.util.Set;
 
-import static org.hamcrest.Matchers.allOf;
-import static org.hamcrest.Matchers.hasEntry;
+import static org.hamcrest.Matchers.equalTo;
 
 public class EsqlResolvedIndexExpressionIT extends AbstractCrossClusterTestCase {
 
     public void testLocalIndices() {
         createIndex(LOCAL_CLUSTER, "index-1");
 
-        assertThat(resolveIndices("index-1"), hasEntry(LOCAL_CLUSTER, resolvedIndexExpression("index-1", "index-1")));
+        assertThat(resolveIndices("index-1"), equalTo(Map.of(LOCAL_CLUSTER, resolvedIndexExpression("index-1", "index-1"))));
```

Comment:
I am replacing `hasEntry(..)` with `equalTo(Map.of(..))` since the first one silently accepted extra entries.